hybrasyl/Objects/User.cs

```diff
@@ -1260,163 +1188,74 @@ public void SendVisibleGold(Gold gold)
-                        {
-                            HealTarget(castable, tar);
-                            spellCast = true;
-                        }
-                    }
-                    if(spellCast) CastingCost(castable);
-                }
-                else
-                {
-                    this.SendSystemMessage("Having no objective in mind, the spell dissipates.");
-                }
-            }
-        }
-
-        public bool canCast(Castable castable)
-        {
-            var costs = castable.CastCosts.CastCost;
-
-            if (double.TryParse(costs.Stat.Hp, out double hpCostPercent))
+            if (UseCastable(castable, targetCreature))
             {
-                if (hpCostPercent != 0)
-                {
-                    uint reduceHp = (uint)Math.Ceiling(this.MaximumHp * hpCostPercent);
-                    if (reduceHp >= this.Hp) return false;
-                }
-            }
-
-            if (double.TryParse(costs.Stat.Mp, out double mpCostPercent))
-            {
-                if (mpCostPercent != 0)
-                {
-                    uint reduceMp = (uint)Math.Ceiling(this.MaximumMp * mpCostPercent);
-                    if (reduceMp > this.Mp) return false;
-                }
-            }
-
-            if (costs.Gold > this.Gold) return false;
-
-
-            if (costs.Items != null)
-            {
-                foreach (var item in costs.Items)
+                Client.Enqueue(new ServerPacketStructures.Cooldown()
                 {
-                    if (item != null)
-                    {
-                        if (!(Inventory.Contains(item.Value, item.Quantity))) return false;
-                    }
-                }
+                    Length = (uint)castable.Cooldown,
+                    Pane = 1,
+                    Slot = slot
+                }.Packet());
             }
-
-            return true;
+            else
+                SendSystemMessage("It didn't work.");
         }
 
-        public void CastingCost(Castable castable)
+        /// <summary>
+        /// Process the casting cost for a castable. If all requirements were not met, return false.
+        /// </summary>
+        /// <param name="castable">The castable that is being cast.</param>
+        /// <returns>True or false depending on success.</returns>
+        public bool ProcessCastingCost(Castable castable)
         {
             var costs = castable.CastCosts.CastCost;
             uint reduceHp = 0;
             uint reduceMp = 0;
             int removeNumItems = 0;
 
-            if (double.TryParse(costs.Stat.Hp, out double hpCostPercent))
-            {
-                if (hpCostPercent != 0)
-                {
-                    reduceHp = (uint)Math.Ceiling(this.MaximumHp * hpCostPercent);
-                }
-            }
+            // HP cost can be either a percentage (0.25) or a fixed amount (50)
+            if (costs.Stat.Hp.Contains('.'))

```

Comment:
What if i want it to be 1, as in 1.00 without specifying 1.00 in xml?

Reply:
the value taken when loaded will return as 1, even if specified in xml.

Reply:
`1` would be the fixed value 1. `1.00` would be a percentage. It is a string in XML and a string type here. 

Since `1.36` has no meaning as a fixed value (given that HP/MP are `uint`) I'm not sure what the issue is here.